examples/chat.rs

```diff
@@ -157,13 +166,13 @@ fn main() {
 
         loop {
             match swarm.poll().expect("Error while polling swarm") {
-                Async::Ready(Some(_)) => {
-
-                },
+                Async::Ready(Some(_)) => {},
                 Async::Ready(None) | Async::NotReady => {
                     if !listening {
-                        if let Some(a) = Swarm::listeners(&swarm).next() {
-                            println!("Listening on {:?}", a);
+                        for addr in Swarm::listeners(&swarm) {
+                            // Since we used the OS assigned port the underlying TCP socket will not
+                            // be opened before the first `swarm.poll`.
```

Comment:
I don't understand this comment. Whether you use OS assigned ports or not, the socket will always only be opened once the swarm is polled. The point of `listening` is to print "Listening on ..." _once_ and to only print it _after_ the swarm actually started listening on that socket.

Reply:
Originally I wanted to add this comment as I didn't quite understand why/how to implement this and didn't figure out search the examples. To be honest now that you pointed this out the comment does not make any sense for me either. I'll just remove this as I cannot figure out a better way to write it.